pandas/tests/frame/test_dtypes.py

```diff
@@ -372,6 +372,76 @@ def test_astype_str(self):
             expected = DataFrame(['1.12345678901'])
             assert_frame_equal(result, expected)
 
+    def test_astype_dict(self):
```

Comment:
you can test here, but test for ``Series`` as well (e.g. it would have to be a 1-element dictionary)